lib/c-flags.h

```diff
@@ -10,6 +10,14 @@ extern "C" {
 #include <stdint.h>
 #include <sys/types.h>
 
+#if defined(C_FLAGS_BUILD_SHARED) && defined(_MSC_VER)
+#define C_FLAGS_EXPORT extern __declspec(dllexport)
+#elif defined(C_FLAGS_BUILD_SHARED) && !defined(_MSC_VER)
+#define C_FLAGS_EXPORT __attribute__((visibility("default")))
```

Comment:
Note that technically you probably want to check for `__GNUC__` support (not all unix compilers support GCC extensions, though the common ones do).

Also, Windows gets upset if you use dllexport in a non-shared build, but GCC just does the right thing with default visibility, so it is safe to unconditionally define the visibility attribute and can produce better codegen optimization opportunities even for static builds.

Reply:
Hi, do you suggest doing something like this?

```c++
#if defined(C_FLAGS_MSVC_BUILD_SHARED)
#define C_FLAGS_EXPORT extern __declspec(dllexport)
#elif defined(__GNUC__)
#define C_FLAGS_EXPORT __attribute__((visibility("default")))
#else
#define C_FLAGS_EXPORT
#endif
```

And in the `meson.build` file do something like this?

```meson
build_args = []

if get_option('default_library') == 'shared' and compiler.get_id() == 'msvc'
    build_args += '/DC_FLAGS_MSVC_BUILD_SHARED'
endif
```

Reply:
compiler.get_argument_syntax() maybe -- that should also handle clang-cl as a mostly dropin msvc replacement.

(You can either use _MSC_VER in the header or do the compiler check in meson.build before defining the build_args, either one should work okay.)

Reply:
@eli-schwartz, if interested, you can see the changes in PR #2.